boards/arc/nsim_em/doc/board.rst

```diff
@@ -0,0 +1,79 @@
+.. _nsim_em:
+
+DesignWare(R) ARC(R) Emulation (nsim)
+#####################################
+
+Overview
+********
+
+This board configuration will use `Designware ARC nSIM`_ to emulate a virtual ARC EM based board including the following features:
+
+* ARC EM processor
+* ARC internal timer
+* a virtual output only console (uart-nsim)
+
+There are two sub configurations in board:
+
+* nsim_em which includes normal em features and ARC MPUv2
+* nsim_sem which includes secure em features and ARC MPUv3
+
+For detailed arc features, please refer :file:`boards/arc/nsim_em/support/nsim_em.props` and :file:`boards/arc/nsim_em/support/nsim_sem.props`.
+
+
+Hardware
+********
+Supported Features
+==================
+
+The following hardware features are supported:
+
++-----------+------------+-----+-------+-----------------------+
+| Interface | Controller | EM  | SEM   | Driver/Component      |
++===========+============+=====+=======+=======================+
+| INT       | on-chip    | Y   | Y     | interrupt_controller  |
++-----------+------------+-----+-------+-----------------------+
+| UART      | nsim uart  | Y   | Y     | serial port-polling   |
++-----------+------------+-----+-------+-----------------------+
+| TIMER     | on-chip    | Y   | Y     | system clock          |
++-----------+------------+-----+-------+-----------------------+
+
+
+Programming and Debugging
+*************************
+
+Required Hardware and Software
+==============================
+
+To use Zephyr RTOS applications on this board, `Designware ARC nSIM`_ or `Designware ARC nSIM Lite`_ is required.
+
+Building Sample Applications
+==============================
+
+Use this configuration to run basic Zephyr applications and kernel tests in nSIM, for example, with the :ref:`synchronization_sample`:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/synchronization
+   :host-os: unix
+   :board: nsim_em
+   :goals: flash
```

Comment:
should be 'make run', btw, how do you get out of the simulator? I can't abort it for some reason, what is magic key combo? :-)

Reply:
The CTRL-] combination redirects input from the simulated UART back to the simulator, and CTRL-C stops the simulation. You might need to type a 'reset' command to get the terminal behave normal again, not sure what magic the UART simulation does.

Another trick: the nsimdrv command invoked in the nsim runner has a timeout parameter "-s 100000000" stops the simulation after 100M cycles.

Reply:
Because,  the virtual uart in nsim will hijack the stdin of console. So "ctrl+c" will be caught by the virtual uart,  not the console(nsimdrv). On the other hand, it seems that the nsim has no direct support of "ctrl+c", i.e., if the nsim is running, no way to pause it by "ctrl+c", no magic key combo found.   

So one way to debug is:  open two consoles, one console for make debugserver, the other console to run xxx-xxx-gdb.